src/herbivoresAndCarnivores.js

```diff
@@ -2,14 +2,47 @@
 
 class Animal {
   // write your code here
+
+  constructor(name, health = 100) {
+    this.health = health;
+    this.name = name;
+    Animal.addAnimal(this);
+  }
+
+  static addAnimal(animal) {
+    Animal.alive.push(animal);
+  }
+
+  static removeAnimal(animal) {
+    Animal.alive = Animal.alive.filter(item => item.name !== animal.name);
+  }
 }
 
+Animal.alive = [];
+
 class Herbivore extends Animal {
   // write your code here
+  constructor() {
+    super();
+    this.hidden = false;
+  }
+
+  hide() {
+    this.hidden = !this.hidden;
+  }
 }
 
 class Carnivore extends Animal {
   // write your code here
+  bite(herbivore) {
+    if (herbivore.hidden === false) {
```

Comment:
you should check if it's in fact Herbivore, use `instance of` for that

Reply:
also, it's easier to check boolean values by using logical NO
```suggestion
    if (!herbivore.hidden) {
```